src/TagLib.Shared/TagLib/StreamFileAbstraction.cs

```diff
@@ -18,29 +18,121 @@
 // along with this library.
 //
 
+using System;
 using System.IO;
 
 namespace TagLib
 {
     public class StreamFileAbstraction : File.IFileAbstraction
     {
-        public StreamFileAbstraction(string name, Stream readStream, Stream writeStream)
+        /// <summary>
+        /// Initializes a StreamFileAbstaction with a Byte[] of data for the media
+        /// Example usage:
+        /// 
+        ///                var abs = new StreamFileAbstraction(filename, songData);
+        ///                using (var file = File.Create(abs))
+        ///                {
+        ///                    file.Tag.Album = Album;
+        ///                    file.Tag.Title = SongTitle;
+        ///                    file.Tag.AlbumArtists = new[] {Artist};
+        ///                    file.Tag.AmazonId = Song.AmazonTrackId;
+        ///                    file.Save();
+        ///                }
+        ///                StorageFile sf = await folder.CreateFileAsync(filename, CreationCollisionOption.FailIfExists);
+        ///                sf.WriteAllBytes(abs.TaggedMediaData); 
+        /// 
+        /// </summary>
+        /// <param name="name"> Name of the media file</param>
+        /// <param name="data"> Byte Array of the media source</param>
+        public StreamFileAbstraction(string name, byte[] data)
+        {
+            ReadStream = new MemoryStream();
+            ReadStream.Write(data, 0, data.Length);
+            WriteStream = ReadStream;
+            Name = name;
+        }
+
+        public StreamFileAbstraction(string name, MemoryStream readStream)
```

Comment:
Yes, I guess it could be considered that, or it could be considered an improvement because it would prevent others from having the problem I had when I created a memory stream to use and initialized it using a byte[] thus making not expandable.